core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java

```diff
@@ -707,6 +707,12 @@ public AnnotationClassValue<?>[] annotationClassValues(@NonNull String member) {
         if (o instanceof AnnotationClassValue<?>[] annotationClassValues) {
             return annotationClassValues;
         }
+        if (o instanceof String className) {
+            return new AnnotationClassValue[] { getAnnotationClassValue(className) };
+        }
+        if (o instanceof String[] classNames) {
+            return Arrays.stream(classNames).map(AnnotationValue::getAnnotationClassValue).toArray(AnnotationClassValue[]::new);
+        }
```

Comment:
Please also add the same logic to the method below `public Optional<AnnotationClassValue<?>> annotationClassValue(@NonNull String member) {`

Reply:
Singular method to create `AnnotationClassValue` is used for `@Requires(bean = Class)` or `@Requires(condition = Class)` and can't get class names. Only `@Requires` field possibly taking strings is `classNames`: 
```
@AliasFor(member = "missing")
String[] missingClasses() default {};
```

I can add the same change but can't test it because only case when `@Requires` takes in string(s) is `missingClasses` and it is covered above.

Reply:
Please add it and add a test to `AnnotationValueSpec`

Reply:
Done in https://github.com/micronaut-projects/micronaut-core/pull/11087/commits/b771aca05739eb6255fa84f6320dc80780681751